tests/unit_tests/test_geometry.py

```diff
@@ -203,6 +204,83 @@ def test_get_by_name():
     assert not univs
 
 
+def test_cyl_prism():
+    cyl_prism = openmc.model.cylindrical_prism(2,
+                                               5,
+                                               origin=(-1,1,3))
+    rounded_cyl_prism = openmc.model.cylindrical_prism(2,
+                                                       5,
+                                                       origin=(-1,1,3),
+                                                       upper_fillet_radius=1.6,
+                                                       lower_fillet_radius=1.6)
+    prisms = [cyl_prism, rounded_cyl_prism]
+
+    def make_geometry(prism):
+        mat = openmc.Material()
+        mat.add_nuclide('H1', 1.0)
+        materials = openmc.Materials(materials=[mat])
+
+        cell = openmc.Cell(region=prism)
+        geo = openmc.Geometry([cell])
+
+        settings = openmc.Settings()
+        settings.particles = 1000
+        settings.inactive = 10
+        settings.batches = 50
+
+        openmc.model.Model(settings=settings,
+                           materials=materials,
+                           geometry=geo).export_to_xml()
+
+    # clear checks
+    inside_points = [[(-1, 1, 3),   #center
+                      (-1, 0.5, 1), # lower
+                      (0.5, 1, 5),  # upper
+                      (-1, 2.99, 3), # edges
+                      (0.99, 1, 3),
+                      (-2.4, 1, 5.49),
+                      (-2.4, 1, 0.51)],
+                     [(-1, 1, 3),   #center
+                      (-1, 0.5, 1), # lower
+                      (0.5, 1, 5),
+                      (-1, 2.99, 3),
+                      (0.99, 1, 3)]]
+
+
+    outside_points = [[(-1, 1, 6),
+                       (-1, 1, 0),
+                       (-2, -1, 3),
+                       (-1, 3.01, 3), # edges
+                       (1.01, 1, 3),
+                       (-2.4, 1, 5.51),
+                       (-2.4, 1, 0.49)],
+                      [(-1, 1, 6),
+                       (-1, 1, 0),
+                       (-2, -1, 3),
+                       (-1, 3.01, 3),
+                       (1.01, 1, 3),
+                       (-2.4, 1, 5.51),
+                       (-2.4, 1, 5.49),
+                       (-2.4, 1, 0.49),
+                       (-2.4, 1, 0.51)]]
+
+    openmc.lib.init()
+    for prism, inside_points, outside_points in zip(prisms, inside_points, outside_points):
```

Comment:
There are a few things to address here:

  - When `openmc.lib.init` is called, it will read the XML files and initialize OpenMC in memory, so any subsequent XML files that are generated will be ignored. The `init` and `finalize` calls should be inside of the loop so OpenMC is started and shutdown for each prism being tested here. I'd even look into the `openmc.lib.run_in_memory()` context manager too to simplify this further.
  - The name `cell` is going to be undefined in the loop here. Might be good to return the cell created from the `make_geometry` function.
  - Going to need a boundary condition on at least one surface in the geometry.
  - Need to wrap the string passed to `match` in `pytest.raises` in an `re.escape` call to ask the regex matching not to treat the parentheses characters as a pattern. I've been hit by that one before.

Reply:
Okay, got it working!